### PR TITLE
Allow certain or all roles to be removed upon mute and readded on unmute

### PR DESCRIPTION
**Pull request by DarkView**
All credit goes to @roflmaoqwerty, i merely updated it to work with the new version due to time constraints on his side.
Originally PR #61

### CODE REVIEW

File: backend/src/plugins/Mutes/functions/clearExpiredMutes.ts [hunk 1]
```diff
@@ -14,6 +15,13 @@ export async function clearExpiredMutes(pluginData: GuildPluginData<MutesPluginT
         if (muteRole) {
           await member.removeRole(muteRole);
         }
+        if (mute.roles_to_restore) {
+          const memberOptions: MemberOptions = {};
+          memberOptions.roles = Array.from(
+            new Set([...mute.roles_to_restore, ...member.roles.filter(x => x !== muteRole)]),
+          );
+          member.edit(memberOptions);
```
Comment (Dragory) [inline backend/src/plugins/Mutes/functions/clearExpiredMutes.ts:23]: Might need some validation here to make sure each role we try to restore actually exists

File: backend/src/plugins/Mutes/functions/unmuteUser.ts [hunk 2]
```diff
@@ -36,6 +36,13 @@ export async function unmuteUser(
       if (muteRole && member.roles.includes(muteRole)) {
         await member.removeRole(muteRole);
       }
+      if (existingMute?.roles_to_restore) {
+        const memberOptions: MemberOptions = {};
+        memberOptions.roles = Array.from(
+          new Set([...existingMute.roles_to_restore, ...member.roles.filter(x => x !== muteRole)]),
+        );
+        member.edit(memberOptions);
```
Comment (Dragory) [inline backend/src/plugins/Mutes/functions/unmuteUser.ts:44]: Same validation as above to make sure the roles we're restoring actually exist here

Comment (DarkView) [reply backend/src/plugins/Mutes/functions/unmuteUser.ts:45]: Fixed with 499853b

File: backend/src/plugins/Mutes/functions/clearExpiredMutes.ts (hunk 1 above)
Comment (DarkView) [reply backend/src/plugins/Mutes/functions/clearExpiredMutes.ts:24]: Fixed with 499853b

**Pull request merged**